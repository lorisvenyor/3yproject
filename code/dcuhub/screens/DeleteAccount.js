
import React from 'react'
import { StyleSheet, Text, View, Image, Dimensions, TextInput, Button, TouchableOpacity, Alert, LogBox} from 'react-native'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import Icon from '@expo/vector-icons/AntDesign'
import { StatusBar } from 'expo-status-bar'
import {SafeAreaView} from 'react-native-safe-area-context'
import * as firebase from 'firebase'

const DeviceHeight = Dimensions.get('screen').height

export default class DeleteAccount extends React.Component{
    
    state = {
        email: "",
        password: "",
        password2: "",
        errorMessage: null
    }

    handleAuth = () => {

        const {password,password2} = this.state

        const user = firebase.auth().currentUser
        const credential = firebase.auth.EmailAuthProvider.credential(user.email, password)

        // exception handling
        if(password == ""){
            Alert.alert("", "Please Enter a password.")
        } else if (password2 == ""){
            Alert.alert("","Please confirm your password.")
        }else if (password != password2){
            Alert.alert("","Passwords do not match.")
        } else{

        // Deleting user profile in database upon account deletion
        firebase.firestore().collection('userProfile').doc(user.uid).delete()

        user.reauthenticateWithCredential(credential)
        .catch(error => this.setState({ errorMessage: error.message}))
        
        firebase.auth().currentUser.delete().then(function() {
            console.log('user has deleted its account')
          }).catch(function(error) {
            console.log(error)
          })
        }
    }

    // for when user deletes account
    componentDidMount() {
        LogBox.ignoreLogs(["The user's credential is no longer valid. The user must sign in again."])
    }
    
    render() {
        return (
            <SafeAreaView style={styles.container}>
            <StatusBar style="auto"/>
            <KeyboardAwareScrollView>

                    {/* enter a password */}
                    <View style={{
                        flexDirection:"row",
                        alignItems:"center",
                        marginHorizontal:40,
                        borderWidth: 1.5,
                        marginTop: 150,
                        paddingHorizontal:10,
                        borderColor:"#25afcd",
                        borderRadius:10,
                        paddingVertical: 7
                    }}>
                        <Icon name="lock1" color="#25afcd" size={25}/>
                        <TextInput 
                            style={styles.placeholder_text}
                            secureTextEntry={true}  // SECURE PASSWORD ie hides the texts
                            placeholder="enter password"
                            onChangeText={password => this.setState({password})}
                            value={this.state.password}>
                        </TextInput>
                    </View>
                
                    {/* this is for confirming the password */}
                    <View style={{
                        flexDirection:"row",
                        alignItems:"center",
                        marginHorizontal:40,
                        borderWidth: 1.5,
                        marginTop:15,
                        paddingHorizontal:10,
                        borderColor:"#25afcd",
                        borderRadius:10,
                        paddingVertical: 7
                    }}>
                        <Icon name="lock1" color="#25afcd" size={25}/>
                        <TextInput 
                            style={styles.placeholder_text}
                            secureTextEntry={true}  // SECURE PASSWORD ie hides the texts
                            placeholder="confirm password"
                            autoCapitalize="none"
                            onChangeText={password2 => this.setState({password2})}
                            value={this.state.password2}
                            >
                        </TextInput>
                    </View>

                    <TouchableOpacity onPress={this.handleAuth}>
                        <View style={styles.confirm_button}>
                            <Text style={{color: "#f4a446", fontSize: 20}}>CONFIRM</Text>
                        </View>
                    </TouchableOpacity>
            
            </KeyboardAwareScrollView>
            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    container : {
        flex: 1,
        height: DeviceHeight,
        backgroundColor: "white"
    },
    pass_word: {
        flexDirection:"row",
        alignItems:"center",
        marginHorizontal:40,
        borderWidth: 1.5,
        marginTop:15,
        paddingHorizontal:10,
        borderColor:"#25afcd",
        borderRadius:10,
        paddingVertical: 7,
    },
    placeholder_text: {
        paddingHorizontal: 10
    }, 
    forgot_pass: {
        marginTop: 10,
        alignSelf: 'center'
    },
    confirm_button: {
        marginHorizontal: 120,
        alignItems: "center",
        justifyContent: "center",
        marginTop: 40,
        borderWidth: 2,
        borderColor: "#f4a446",
        paddingVertical: 5,
        borderRadius: 20
    }
})

// colours 
// blue #25afcd
// orange #f4a446
// pink #d40059