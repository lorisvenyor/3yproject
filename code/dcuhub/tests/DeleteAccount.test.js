import React from 'react'
import {create, act} from 'react-test-renderer'
import DeleteAccount from '../screens/DeleteAccount'


const screen = create(<DeleteAccount/>)

test('Delete Account Screen Snapshot', () => {
    expect(screen).toMatchSnapshot()
})
