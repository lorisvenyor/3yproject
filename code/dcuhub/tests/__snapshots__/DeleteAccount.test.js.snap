// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Account Screen Snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "marginHorizontal": 40,
            "marginTop": 150,
            "paddingHorizontal": 10,
            "paddingVertical": 7,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="enter password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "marginHorizontal": 40,
            "marginTop": 15,
            "paddingHorizontal": 10,
            "paddingVertical": 7,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="confirm password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#f4a446",
              "borderRadius": 20,
              "borderWidth": 2,
              "justifyContent": "center",
              "marginHorizontal": 120,
              "marginTop": 40,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#f4a446",
                "fontSize": 20,
              }
            }
          >
            CONFIRM
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
